### PR TITLE
Use File Descriptor for loading presets

### PR DESCRIPTION
**Pull request by dimitry-**
Test: Start Orbit -> Load presets

### CODE REVIEW

**Comment by dimitry-** [comment]
(forgot to fix the saving presets part). working on it.

File: src/OrbitGl/App.cpp [hunk 1]
```diff
@@ -691,36 +691,39 @@ ErrorMessageOr<void> OrbitApp::SavePreset(const std::string& filename) {
...
-  std::ofstream file(filename_with_ext, std::ios::binary);
-  if (file.fail()) {
-    ERROR("Saving preset in \"%s\": %s", filename_with_ext, "file.fail()");
-    return ErrorMessage(
-        absl::StrFormat("Error opening the file \"%s\" for writing", filename_with_ext));
+  auto open_result = orbit_base::OpenFileForWriting(filename_with_ext);
+  if (open_result.has_error()) {
+    ERROR("Failed to open \"%s\": %s", filename_with_ext, open_result.error().message());
+    return ErrorMessage(absl::StrFormat("Failed to open \"%s\": %s", filename_with_ext,
+                                        open_result.error().message()));
```
Comment (IrinaShkviro) [inline src/OrbitGl/App.cpp:698]: Maybe define a message and use it in ERROR and ErrorMessage (as it's done in line 716)? There also similar cases below

**Review by IrinaShkviro — APPROVED** [review]
Thank Dimitry! I've left one comment below

File: src/OrbitGl/App.cpp (hunk 1 above)
Comment (dimitry-) [reply src/OrbitGl/App.cpp:698]: done.

**Pull request merged**